sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/StorageSpec.groovy

```diff
@@ -36,6 +37,15 @@ class StorageSpec extends Specification {
         return namer
     }
 
+    protected <T> T instrument(T builder) {
```

Comment:
I wish we had `interface ClientBuilder` or `abstract class ClientBuilderBase`

Reply:
We actually did have this early on. I don't remember why we opted against it, but I think it may be related to the fact that all the methods return the ClientBuilder type and that was getting kinda messy?

Reply:
well, we could have done:
```
interface ClientBuilder<T extends ClientBuilder> {
   T httpClient(...);
}
```